src/Juvix/Backends/LLVM/JIT/Execution.hs

```diff
@@ -0,0 +1,65 @@
+module Juvix.Backends.LLVM.JIT.Execution
+  ( jit,
+  )
+where
+
+import Control.Concurrent
+import qualified Data.ByteString.Char8 as B
+import Juvix.Backends.LLVM.JIT.Types
+import Juvix.Library
+import qualified LLVM.AST as AST
+import LLVM.Context
+import qualified LLVM.ExecutionEngine as EE
+import LLVM.Module as Mod
+import LLVM.PassManager
+import LLVM.Target
+
+runJIT ∷ Config → Context → (EE.MCJIT → IO a) → IO a
+runJIT config ctx = EE.withMCJIT ctx optlevel model ptrelim fastins
+  where
+    optlevel = convOptLevel (configOptimisationLevel config)
+
+    model = Nothing -- code model (default)
+
+    ptrelim = Nothing -- frame pointer elimination
+
+    fastins = Nothing -- fast instruction selection, apparently not yet supported
+
+passes ∷ Config → PassSetSpec
+passes config = defaultCuratedPassSetSpec {optLevel = convOptLevel (configOptimisationLevel config)}
+
+convOptLevel ∷ OptimisationLevel → Maybe Word
+convOptLevel None = Nothing
+convOptLevel O0 = pure 0
+convOptLevel O1 = pure 1
+convOptLevel O2 = pure 2
+convOptLevel O3 = pure 3
+
+-- Note: in order to allow this to return functions, a green thread is forked to retain module
```

Comment:
return functions?

As in, return newely defined functions that are used at the top level?

Reply:
Return Haskell functions which can then be passed around and used anywhere, yes.